src/features/legacy/Fallback.tsx

```diff
@@ -1,33 +1,38 @@
 import { useEffect, useState } from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
+import { isInsideIframe } from '../../common';
 import PageNotFound from '../layout/PageNotFound';
 
 /**
  * 404s from the legacy site are redirected from legacy.DOMAIN/[some/path/here] to DOMAIN/fallback/[some/path/here]
  * this component handles these fallback redirects for pages such as '/narratives'
  */
 export const Fallback = () => {
-  const location = useParams();
   const navigate = useNavigate();
+  const params = useParams();
 
-  const fallbackPath = location['*']?.toLowerCase();
+  const fallbackPath = params['*']?.toLowerCase();
   const [notFound, setNotFound] = useState(false);
 
   useEffect(() => {
     // Make sure we are in window.top (not within an iframe),
-    if (window && window.top && window !== window.top) {
+    if (window.top && isInsideIframe(window)) {
       // Not in top window, redirect top window to current location
       window.top.location = window.location;
+      return;
+    }
+
+    // We are in the top window
+    // redirect appropriately based on the fallbackPath, otherwise set notFound to true
+    if (fallbackPath === 'narratives') {
+      navigate('/narratives');
+    } else if (Object.hasOwn(params, 'wsId')) {
```

Comment:
Maybe we can use route matching here, instead of relying on string equality or params? Let me find the doc link